Flaky test: `test_component_choice_random`

It failed in this run https://github.com/pymc-devs/pymc/actions/runs/3240831110/jobs/5311952415

With 

```
___________________ TestMixture.test_component_choice_random ___________________

self = <pymc.tests.distributions.test_mixture.TestMixture object at 0x7fb12dd6a2f0>

    def test_component_choice_random(self):
        """Test that mixture choices change over evaluations"""
        with Model() as m:
            weights = [0.5, 0.5]
            components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
            mix = Mixture.dist(weights, components)
        draws = draw(mix, draws=10)
        # Probability of coming from same component 10 times is 0.5**10
>       assert np.unique(draws > 0).size == 2
E       assert 1 == 2
E        +  where 1 = array([False]).size
E        +    where array([False]) = <function unique at 0x7fb14babd360>(array([-10.01041454, -10.00804946, -10.02323384, -10.01368128,\n        -9.99331916,  -9.9980023 ,  -9.98098861, -10.00484836,\n        -9.993316  , -10.00568352]) > 0)
E        +      where <function unique at 0x7fb14babd360> = np.unique

pymc/tests/distributions/test_mixture.py:338: AssertionError
```

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -329,17 +329,13 @@ class TestMixture(SeededTest):
 
     def test_component_choice_random(self):
         """Test that mixture choices change over evaluations"""
-        p_first_component = 0.5
-        n_draws = 200
         with Model() as m:
-            weights = [p_first_component, 1.0 - p_first_component]
+            weights = [0.5, 0.5]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=n_draws, random_seed=self.get_random_state())
-        k_first_component = np.sum(draws < 0)
-        # The number of draws from the first component should follow a binomial distribution
-        result_binomtest = st.binomtest(k_first_component, n_draws, p_first_component)
-        assert result_binomtest.pvalue > 0.01
+        draws = draw(mix, draws=10, random_seed=self.get_random_state())
+        # Probability of coming from same component 10 times is 0.5**10
+        assert np.unique(draws > 0).size == 2
 
     @pytest.mark.parametrize(
         "comp_dists",

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -329,13 +329,17 @@ class TestMixture(SeededTest):
 
     def test_component_choice_random(self):
         """Test that mixture choices change over evaluations"""
+        p_first_component = 0.5
+        n_draws = 200
         with Model() as m:
-            weights = [0.5, 0.5]
+            weights = [p_first_component, 1.0 - p_first_component]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=10, random_seed=self.get_random_state())
-        # Probability of coming from same component 10 times is 0.5**10
-        assert np.unique(draws > 0).size == 2
+        draws = draw(mix, draws=n_draws, random_seed=self.get_random_state())
+        k_first_component = np.sum(draws < 0)
+        # The number of draws from the first component should follow a binomial distribution
+        result_binomtest = st.binomtest(k_first_component, n_draws, p_first_component)
+        assert result_binomtest.pvalue > 0.01
 
     @pytest.mark.parametrize(
         "comp_dists",

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -334,7 +334,7 @@ class TestMixture(SeededTest):
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
         draws = draw(mix, draws=20, random_seed=self.get_random_state())
-        # Probability of coming from same component 200 times is 0.5**200
+        # Probability of coming from same component 20 times is 0.5**20
         assert np.unique(draws > 0).size == 2
 
     @pytest.mark.parametrize(

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -333,7 +333,7 @@ class TestMixture(SeededTest):
             weights = [0.5, 0.5]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=10)
+        draws = draw(mix, draws=10, random_seed=self.get_random_state())
         # Probability of coming from same component 10 times is 0.5**10
         assert np.unique(draws > 0).size == 2
 

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -333,7 +333,7 @@ class TestMixture(SeededTest):
             weights = [0.5, 0.5]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=200, random_seed=self.get_random_state())
+        draws = draw(mix, draws=20, random_seed=self.get_random_state())
         # Probability of coming from same component 200 times is 0.5**200
         assert np.unique(draws > 0).size == 2
 

--- a/pymc/tests/distributions/test_mixture.py
+++ b/pymc/tests/distributions/test_mixture.py
@@ -333,8 +333,8 @@ class TestMixture(SeededTest):
             weights = [0.5, 0.5]
             components = [Normal.dist(-10, 0.01), Normal.dist(10, 0.01)]
             mix = Mixture.dist(weights, components)
-        draws = draw(mix, draws=10, random_seed=self.get_random_state())
-        # Probability of coming from same component 10 times is 0.5**10
+        draws = draw(mix, draws=200, random_seed=self.get_random_state())
+        # Probability of coming from same component 200 times is 0.5**200
         assert np.unique(draws > 0).size == 2
 
     @pytest.mark.parametrize(

